### PR TITLE
Add support to enable and disable authenticators

### PR DESCRIPTION
**Pull request by rafis3**
### Desired Outcome

The Conjur Python API library should have a newly supported action of enabling and disabling of an authenticator through REST API.

### Implemented Changes

- Added .gitignore file to the repository
- A new action was added in the client to expose the new functionality
- A new action was added in the API implementation to support the new functionality
- New unit and integration tests have been added for this functionality
- Update the test base image to Ubuntu 22.04

### Connected Issue/Story

ONYX-23423

### Definition of Done

- Python SDK supports enabling an authenticator
- Python SDK supports disabling an authenticator

#### Changelog

- [x] The CHANGELOG has been updated

#### Test coverage

- [x] This PR includes new unit and integration tests to go with the code
  changes, or

#### Documentation

- [x] Docs (e.g. `README`s) were updated in this PR

#### Behavior

- [x] No behavior was changed with this PR, but new functionality was added

#### Security

- [x] There are no security aspects to these changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -215,6 +215,14 @@ _Note: This method requires Conjur v1.9+_
 Returns a Python dictionary of information about the client making an API request (such as its IP address, user,
 account, token expiration date, etc).
 
+#### `set_authenticator_state(authenticator_id, enabled)`
+
+Allows enabling and disabling an authenticator.  
+
+_Note: This functionality relies on an endpoint in Conjur which is part of an early implementation of support for
```
Comment (egvili) [inline README.md:222]: Is there a more mature endpoint that we can use instead?

File: conjur_api/http/api.py [hunk 2]
```diff
@@ -504,6 +504,26 @@ async def rotate_personal_api_key(
...
+        """
+        This method sets the state of a given authenticator
+        """
+
+        params = {
+            'authenticator_id': authenticator_id
+        }
+        params.update(self._default_params)
+
+        body = f'enabled={"true" if enabled else "false"}'
```
Comment (egvili) [inline conjur_api/http/api.py:517]: Maybe we can use `str(enabled).lower()` to get the same result. 
Not sure what is better, just throwing in idea

File: tests/integration/test_integration_authenticator_state.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+    @classmethod
+    def setUpClass(cls):
+        asyncio.run(cls._add_test_data())
+
+    async def test_authenticator_enable_success(self):
+        c = await self._create_client(self.valid_user.id, self.valid_user.api_key)
+        response = await c.set_authenticator_state('authn-iam/test', True)
+
+        self.assertEqual(response, '')
```
Comment (egvili) [inline tests/integration/test_integration_authenticator_state.py:31]: Empty response means success, right?

File: tests/integration/test_integration_authenticator_state.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        invalid_user: ConjurUser = ConjurUser(user_id='test-authenticator-invalid-user',
+                                              api_key=response['created_roles']['dev:user:test-authenticator-invalid-user']
+                                              ['api_key'])
+        response = await c.load_policy_file('conjur', 'tests/integration/policies/authn-iam-test.yml')
+
+        cls.valid_user = valid_user
+        cls.invalid_user = invalid_user
+
+    @classmethod
+    async def _create_client(cls, username: str, password: str) -> Client:
```
Comment (egvili) [inline tests/integration/test_integration_authenticator_state.py:81]: Just for understanding: does that function create real SDK client, but it does not interact with real conjur?
How does this magic happen?

**Review by egvili — APPROVED** [review]
LGTM. 
A few questions/suggestion inside

File: tests/integration/test_integration_authenticator_state.py (hunk 3 above)
Comment (rafis3) [reply tests/integration/test_integration_authenticator_state.py:31]: Right. That's how the REST API works as well, with a 204 HTTP code. But the code is not usually returned when it's valid.

File: tests/integration/test_integration_authenticator_state.py (hunk 4 above)
Comment (rafis3) [reply tests/integration/test_integration_authenticator_state.py:81]: It is interacting with a real Conjur. The integration test runs a local Conjur environment and then runs the tests against it. See the `ubuntu_compose.yml` file located in `ci/test/conjur-deployment` for more details

File: conjur_api/http/api.py (hunk 2 above)
Comment (rafis3) [reply conjur_api/http/api.py]: Yes, I like it. Fixed, thanks!

File: tests/integration/test_integration_authenticator_state.py (hunk 4 above)
Comment (egvili) [reply tests/integration/test_integration_authenticator_state.py:81]: Ohh now I understand :)

**Pull request merged**